set rust crate lto to thin to not use fatlto for gkrust which ooms on 32-bit.
also a lot faster
diff --git a/config/makefiles/rust.mk b/config/makefiles/rust.mk
index ab3304484180..5c717eb0850a 100644
--- a/config/makefiles/rust.mk
+++ b/config/makefiles/rust.mk
@@ -56,7 +56,7 @@ cargo_build_flags += $(MOZ_CARGO_BUILD_STD_ARGS)
 cargo_rustc_flags = $(CARGO_RUSTCFLAGS)
 # Enable link-time optimization for release builds.
 ifdef RUST_LIBRARY_LTO
-cargo_rustc_flags += -Clto
+cargo_rustc_flags += -Clto=thin
 endif
 
 ifdef CARGO_INCREMENTAL
